File: in_Java/0303 Range Sum Query - Immutable.java
class NumArray {
    int[] cursum;
    public NumArray(int[] nums) {
        cursum = new int[nums.length+1];
        for(int i = 1; i < cursum.length; i++){
            cursum[i] = cursum[i-1] + nums[i-1];
        }
    }
    
    public int sumRange(int i, int j) {
        return cursum[j+1] - cursum[i];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(i,j);
 */

File: in_Python/0303 Range Sum Query - Immutable.py
class NumArray(object):

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.arr = nums
        for i in range(1, len(nums)):
            self.arr[i] += self.arr[i-1]
        self.arr.insert(0,0)

    def sumRange(self, i, j):
        """
        :type i: int
        :type j: int
        :rtype: int
        """
        return self.arr[j+1] - self.arr[i]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)